Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/AuditDBStoreImpl.java

```diff
@@ -0,0 +1,131 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.auditrepository.nhinc;
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import com.services.nhinc.schema.auditmessage.ObjectFactory;
+import gov.hhs.fha.nhinc.auditrepository.hibernate.AuditRepositoryDAO;
+import gov.hhs.fha.nhinc.auditrepository.hibernate.AuditRepositoryRecord;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventSecureRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.sql.Blob;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.datatype.XMLGregorianCalendar;
+import org.hibernate.Hibernate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author tjafri
+ */
+public class AuditDBStoreImpl implements AuditStore {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditDBStoreImpl.class);
+    private static final AuditRepositoryDAO auditLogDao = AuditRepositoryDAO.getAuditRepositoryDAOInstance();
+
+    @Override
+    public boolean saveAuditRecord(LogEventSecureRequestType request, AssertionType assertion) {
+        List<AuditRepositoryRecord> auditRecList = new ArrayList<>();
+        auditRecList.add(createDBAuditObj(request, assertion));
+        LOG.trace("AuditRepositoryOrchImpl.logAudit() -- Calling auditLogDao to insert record into database.");
+        return auditLogDao.insertAuditRepository(auditRecList);
+    }
+
+    protected final AuditRepositoryRecord createDBAuditObj(LogEventSecureRequestType mess, AssertionType assertion) {
+        AuditRepositoryRecord auditRec = new AuditRepositoryRecord();
+        String eventCommunityId = mess.getRemoteHCID();
+        if (NullChecker.isNotNullish(eventCommunityId)) {
+            auditRec.setRemoteHcid(eventCommunityId);
+        } else {
+            auditRec.setRemoteHcid("");
+        }
+
+        auditRec.setDirection(mess.getDirection());
+        auditRec.setMessage(getBlobFromAuditMessage(mess.getAuditMessage()));
+
+        XMLGregorianCalendar xMLCalDate = mess.getEventTimestamp();
+        if (xMLCalDate != null) {
+            auditRec.setEventTimeStamp(convertXMLGregorianCalendarToDate(xMLCalDate));
+        }
+        auditRec.setOutcome(mess.getEventOutcomeIndicator().intValue());
+        auditRec.setEventType(mess.getEventType());
+        auditRec.setEventId(mess.getEventID());
+        auditRec.setMessageId(mess.getRequestMessageId());
+        auditRec.setRelatesTo(mess.getRelatesTo());
+        auditRec.setUserId(mess.getUserId());
+
```

Comment:
Since db can accept null values there is no point in checking for null for these fields.
Note:
A NPE happens only when you access a method on a null object. It does not happen if you are calling a method and getting a null value.